src/overlays/actors/ovl_En_Rr/z_en_rr.c

```diff
@@ -21,51 +81,772 @@ const ActorInit En_Rr_InitVars = {
+            this->bodySegs[i].scale.z = this->bodySegs[i].scale.x;
+            Math_SmoothScaleMaxF(&this->bodySegs[i].height, this->bodySegs[i].heightTarget, 1.0f,
+                                 this->unk_364 * 300.0f);
+        }
+        Math_SmoothScaleMaxF(&this->unk_364, 1.0f, 1.0f, 0.2f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE978C.s")
+void EnRr_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f zeroVec;
+    EnRr* this = THIS;
+    s32 i;
+    Mtx* segMats = Graph_Alloc(globalCtx->state.gfxCtx, 4 * sizeof(Mtx));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE9880.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_rr.c", 1478);
+    if (1) {}
+    func_80093D84(globalCtx->state.gfxCtx);
+    gSPSegment(POLY_XLU_DISP++, 0x0C, segMats);
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (this->scrollTimer * 0) & 0x7F,
+                                (this->scrollTimer * 0) & 0x3F, 32, 16, 1, (this->scrollTimer * 0) & 0x3F,
+                                (this->scrollTimer * -6) & 0x7F, 32, 16));
+    Matrix_Push();
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE98F8.s")
+    Matrix_Scale((1.0f + this->bodySegs[RR_BASE].scaleMod.x) * this->bodySegs[RR_BASE].scale.x,
+                 (1.0f + this->bodySegs[RR_BASE].scaleMod.y) * this->bodySegs[RR_BASE].scale.y,
+                 (1.0f + this->bodySegs[RR_BASE].scaleMod.z) * this->bodySegs[RR_BASE].scale.z, 1);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_rr.c", 1501),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    Matrix_Pull();
+    VEC_SET(zeroVec, 0.0f, 0.0f, 0.0f);
+    for (i = 1; i < 5; i++) {
+        Matrix_Translate(0.0f, this->bodySegs[i].height + 1000.0f, 0.0f, 1);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE9C88.s")
+        Matrix_RotateRPY(this->bodySegs[i].rot.x, this->bodySegs[i].rot.y, this->bodySegs[i].rot.z, 1);
+        Matrix_Push();
+        Matrix_Scale((1.0f + this->bodySegs[i].scaleMod.x) * this->bodySegs[i].scale.x,
+                     (1.0f + this->bodySegs[i].scaleMod.y) * this->bodySegs[i].scale.y,
+                     (1.0f + this->bodySegs[i].scaleMod.z) * this->bodySegs[i].scale.z, 1);
+        Matrix_ToMtx(segMats, "../z_en_rr.c", 1527);
+        Matrix_Pull();
+        segMats++;
+        Matrix_MultVec3f(&zeroVec, &this->effectPos[i]);
+    }
+    this->effectPos[0] = this->actor.posRot.pos;
+    Matrix_MultVec3f(&zeroVec, &this->mouthPos);
+    gSPDisplayList(POLY_XLU_DISP++, D_06000470);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE9D1C.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_rr.c", 1551);
+    if (this->effectTimer != 0) {
+        static Vec3f effectOffsets[] = {
+            { 25.0f, 0.0f, 0.0f },
+            { -25.0f, 0.0f, 0.0f },
```

Comment:
have to be in function to match?

Reply:
No, but it's where it makes the most sense.

Reply:
i see. i prefer not to put longer pieces of data inside of functions when possible. this one is kinda on the cusp of where id draw the line.
Id probably put it right above the function if it were me, but its probably fine either way

Reply:
I've moved it outside, but wait on merging this. Taking one last look through the code makes me think I might have misnamed a couple fields.

Reply:
OK, fixed names.